### PR TITLE
root: support running authentik in subpath

### LINKED ISSUE #4509: Run Authentik in subfolder behind Traefik Proxy
**Describe your question/**
Due to various reasons I can't run authentik as subdomain (e.g. auth.example.com). I wan't to use authentik as a subfolder under my domain, e.g. example.com/auth/. Additionally I run authentik behind a traefik proxy like I do with many other services, also with subfolders ( example.com/service1/, example.com/service2/ etc.). 

The first call to example.com/auth/ actually reaches authentik, because it redirects me to the first flow (".../flows/-/default/authentication/?next=/") BUT it cuts the subfolder so instead of example.com/auth/flows/-/default/authentication/?next=/  it redirects to example.com/flows/-/default/authentication/?next=/ resulting in a 404 error on traefik. 

In other docker containers there is something like `BASE_URL` or `ROOT_URL`, but I haven't found anything like that. 
I even searched and played around in `authentik/root/settings.py` and in `authentik/root/urls.py` but without success, but I'm a noob in python. 

**Relevant infos**
My traefik labels are:
```
- "traefik.enable=true"
- "traefik.http.routers.authentik-secure.entrypoints=websecure"
- "traefik.http.routers.authentik-secure.tls=true"
# route to subfolder
- "traefik.http.routers.authentik-secure.rule=Host(`example.com`) && PathPrefix(`/auth{regex:$$|/.*}`)"
- "traefik.http.routers.authentik-secure.middlewares=authentik-secure-stripprefix"
- "traefik.http.middlewares.authentik-secure-stripprefix.stripprefix.prefixes=/auth"
- "traefik.http.services.authentik-secure.loadbalancer.server.port=9000"
```
I tried the pathprefix with or without regex and with or without stripprefix. 

**Version and Deployment (please complete the following information):**
 - authentik version: latest
 - Deployment: docker-compose

Thanks for your help in advance!

### PR DESCRIPTION
**Pull request by BeryJu**
## Details

closes #4509

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by BeryJu** [comment]
> There probably are security implications with this, like other apps being able to access authentik's cookies. We should probably mention them in the documentation.

Oh yeah, I had this set when initially testing but it must've gotten removed at some point: https://github.com/goauthentik/authentik/pull/8675/commits/06723cb621376acaf146ccfe8df5f322caee2508

File: internal/config/struct.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ type Config struct {
 	// Config for both core and outposts
 	Debug  bool         `yaml:"debug" env:"AUTHENTIK_DEBUG, overwrite"`
 	Listen ListenConfig `yaml:"listen" env:", prefix=AUTHENTIK_LISTEN__"`
+	Web    WebConfig    `yaml:"web" env:", prefix=AUTHENTIK__WEB__"`
```
Comment (rissson) [inline internal/config/struct.go:17]:
```suggestion
	Web    WebConfig    `yaml:"web" env:", prefix=AUTHENTIK_WEB__"`
```

**Review by rissson — CHANGES_REQUESTED** [review]

**Review by rissson — APPROVED** [review]
Haven't tested it myself, but looks ok. I do wonder what would happen with media files when stored as files though.

**Comment by cfradewavecom** [comment]
I would like to test this but got image `NotFound` when using `ghcr.io/goauthentik/dev-server:gh-3a176a0024e688cd6b126923b221b3e207314def`

Can someone retry pipeline or event better rebase this PR with latest release? Thanks

File: web/src/user/LibraryPage/ApplicationEmptyState.ts [hunk 2]
```diff
@@ -50,7 +50,7 @@ export class LibraryPageApplicationEmptyList extends AKElement {
                 <a
                     aria-disabled="false"
                     class="cta pf-c-button pf-m-secondary"
-                    href="/if/admin/${href}"
+                    href="${globalAK().api.base}if/admin/${href}"
```
Comment (kensternberg-authentik) [inline web/src/user/LibraryPage/ApplicationEmptyState.ts:53]: This repetition starts to feel like it needs to be, oh, I dunno, a _function_ somewhere.  :-)

**Review by kensternberg-authentik — APPROVED** [review]
The Typescript looks fine.  A couple of points come to mind: we have three different URLs: internal, _another interface_, and external.  Supporting this feature makes using a browser-based router that much harder, which in turn will make the command palette a... challenge.  :-)

**Comment by xhb** [comment]
Do you have any updates on this PR? I'm really looking forward to this feature, please merge it as soon as possible.

File: web/src/admin/AdminInterface/AdminSidebar.ts [hunk 3]
```diff
@@ -112,7 +113,7 @@ export class AkAdminSidebar extends WithCapabilitiesConfig(AKElement) {
 
         // prettier-ignore
         const sidebarContent: SidebarEntry[] = [
-            ["/if/user/", msg("User interface"), { "?isAbsoluteLink": true, "?highlight": true }],
+            [`${globalAK().api.base}if/user/`, msg("User interface"), { "?isAbsoluteLink": true, "?highlight": true }],
```
Comment (rissson) [inline web/src/admin/AdminInterface/AdminSidebar.ts:116]: There is also the impersonation trigger where we current redirect to `/`

Comment (BeryJu) [reply web/src/admin/AdminInterface/AdminSidebar.ts:116]: https://github.com/goauthentik/authentik/pull/8675/files#diff-153c71d69a43c3df5e85a6ec763a2ff5e1b7e168ef4975ef1915f13131ad9635 fixed

File: internal/web/proxy.go [hunk 4]
```diff
@@ -32,7 +33,10 @@ func (ws *WebServer) configureProxy() {
 	}
 	rp.ErrorHandler = ws.proxyErrorHandler
 	rp.ModifyResponse = ws.proxyModifyResponse
-	ws.m.PathPrefix("/").HandlerFunc(sentry.SentryNoSample(func(rw http.ResponseWriter, r *http.Request) {
+	ws.mainRouter.PathPrefix(config.Get().Web.Path).Path("/-/health/live/").HandlerFunc(sentry.SentryNoSample(func(rw http.ResponseWriter, r *http.Request) {
+		rw.WriteHeader(204)
```
Comment (rissson) [inline internal/web/proxy.go:37]: Should be 200, unlucky

**Pull request merged**